core/src/main/scala/tech/beshu/ror/configuration/SslConfiguration.scala

```diff
@@ -24,18 +24,20 @@ import io.circe.{Decoder, DecodingFailure, HCursor}
 import monix.eval.Task
 import org.apache.logging.log4j.scala.Logging
 import tech.beshu.ror.configuration.SslConfiguration.{ExternalSslConfiguration, InternodeSslConfiguration, KeystoreFile, TruststoreFile}
+import tech.beshu.ror.providers.{EnvVarsProvider, OsEnvVarsProvider}
 
 import scala.language.implicitConversions
 
 final case class RorSsl(externalSsl: Option[ExternalSslConfiguration],
                         interNodeSsl: Option[InternodeSslConfiguration])
 
-object RorSsl extends LoadFromEsConfig with Logging {
+object RorSsl extends Logging {
 
   val noSsl = RorSsl(None, None)
 
   def load(esConfigFolderPath: Path): Task[Either[MalformedSettings, RorSsl]] = Task {
     implicit val sslDecoder: Decoder[RorSsl] = SslDecoders.rorSslDecoder(esConfigFolderPath)
+    implicit val envVarsProvider:EnvVarsProvider = OsEnvVarsProvider
```

Comment:
IMO we should pass this envprovider as implicit. The implementation of  EnvVarsProvider should be fixed as close to app boundary as possible.